src/Platform/AcquiaCloudPlatform.php

```diff
@@ -182,19 +185,34 @@ public static function getEnvironmentQuestion(Config $config, AcquiaCloudClientF
    * {@inheritdoc}
    */
   public function execute(Command $command, InputInterface $input, OutputInterface $output) : int {
-    $environments = new Environments($this->getAceClient());
     $sites = $this->getPlatformSites();
     if (!$sites) {
       $output->writeln('<warning>No sites available. Exiting...</warning>');
       return 1;
     }
 
+    $group_name = $input->getOption('group');
     $input_uri = $input->getOption('uri');
-    $sites = array_column($sites, 'uri');
-    $args = $this->dispatchPlatformArgumentInjectionEvent($input, $sites, $command);
+
+    if (!$input_uri && $group_name) {
+      $alias = $this->getAlias();
+      $platform_id = self::getPlatformId();
+      $sites = $this->filterSitesByGroup($group_name, $sites, $output, $alias, $platform_id);
+      if (empty($sites)) {
+        return 3;
```

Comment:
Why is this 3 and not 2? Can we add definition to the numbers? Perhaps here as well, saving them into constants with meaningful names?

Reply:
In this case for example:
```php
public const EmptySitesError = 2;
```
Would be much more meaningful, also is there any other place it could be used?

Reply:
Also we already checked for sites in the first portion. We return with 1 if there isn't any. What's the difference between the first and second case? There's no error message to let the user know.

Reply:
My bad there was, it was in the `filterSitesByGroup` function.

Reply:
Which seems to be merged, but I think the method itself does too much. Not to mention the dependencies introduced in the function.